src/Simplify.elm

```diff
@@ -11978,6 +11982,36 @@ callOnEmptyReturnsCheck config collection checkInfo =
             Nothing
 
 
+callOnSingleItemReturnsIt : CallCheckInfo -> Maybe (Error {})
+callOnSingleItemReturnsIt checkInfo =
+    case fullyAppliedLastArg checkInfo of
+        Just lastArg ->
+            case Node.value lastArg of
+                Expression.ListExpr [ Node singleRange _ ] ->
+                    Just
+                        (Rule.errorWithFix
+                            { message = qualifiedToString (qualify checkInfo.fn defaultQualifyResources) ++ " on a singleton list will result in the value inside"
+                            , details = [ "You can replace this call by the value inside the singleton list." ]
+                            }
+                            checkInfo.fnRange
+                            [ Fix.removeRange
+                                { start = checkInfo.parentRange.start
+                                , end = singleRange.start
+                                }
+                            , Fix.removeRange
+                                { start = singleRange.end
+                                , end = checkInfo.parentRange.end
+                                }
+                            ]
```

Comment:
This will e.g. fix
```elm
[ f <| string ] |> String.join sep
--> f <| string
```
which can lead to invalid syntax.
A better alternative is `replaceBySubExpressionFix`, see https://github.com/jfmengels/elm-review-simplify/blob/main/src/Simplify.elm#L14642-L14648

Adding a test for this would be nice, too